core/src/Microsoft.Bot.Core/Hosting/AddBotApplicationExtensions.cs

```diff
@@ -187,19 +187,69 @@ private static IServiceCollection ConfigureMSALWithFIC(this IServiceCollection s
         return services;
     }
 
-    private static IServiceCollection ConfigureMSALFromBotConfig(this IServiceCollection services, BotConfig botConfig)
+    private static IServiceCollection ConfigureMSALWithUMI(this IServiceCollection services, string tenantId, string clientId, string? managedIdentityClientId = null)
+    {
+        ArgumentNullException.ThrowIfNullOrWhiteSpace(tenantId);
+        ArgumentNullException.ThrowIfNullOrWhiteSpace(clientId);
+
+        // Register ManagedIdentityOptions for BotAuthenticationHandler to use
+        bool isSystemAssigned = IsSystemAssignedManagedIdentity(managedIdentityClientId);
+        string? umiClientId = isSystemAssigned ? null : (managedIdentityClientId ?? clientId);
+
+        services.Configure<ManagedIdentityOptions>(options =>
+        {
+            options.UserAssignedClientId = umiClientId;
+        });
+
+        services.Configure<MicrosoftIdentityApplicationOptions>(options =>
+        {
+            options.Instance = "https://login.microsoftonline.com/";
```

Comment:
this means we won't support non public clouds when not using MSAL config scheme.. 

For parity, this is something we could add to our CoreConfig...  

@heyitsaamir any thoughts?